extensions/json-language-features/client/src/languageParticipants.ts

```diff
@@ -0,0 +1,89 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { DocumentSelector } from 'vscode-languageclient';
+import { Event, EventEmitter, extensions } from 'vscode';
+
+/**
+ * JSON language participant contribution.
+ */
+interface LanguageParticipantContribution {
+	/**
+	 * The id of the language which participates with the JSON language server.
+	 */
+	languageId: string;
+	/**
+	 * true if the language allows comments and false otherwise.
+	 * TODO: implement server side setting
+	 */
+	comments?: boolean;
+}
+
+export interface LanguageParticipants {
+	readonly onDidChange: Event<void>;
+	readonly documentSelector: DocumentSelector;
+	hasLanguage(languageId: string): boolean;
+	useComments(languageId: string): boolean;
+	dispose(): void;
+}
+
+export function getLanguageParticipants(): LanguageParticipants {
+	const onDidChangeEmmiter = new EventEmitter<void>();
+	let languages = new Set<string>();
+	let comments = new Set<string>();
+
+	function update() {
+		const oldLanguages = languages, oldComments = comments;
+
+		languages = new Set();
+		languages.add('json');
+		languages.add('jsonc');
+		comments = new Set();
+		comments.add('jsonc');
+
+		for (const extension of extensions.all) {
+			const jsonLanguageParticipants = extension.packageJSON?.contributes?.jsonLanguageParticipants as LanguageParticipantContribution[];
+			if (Array.isArray(jsonLanguageParticipants)) {
+				for (const jsonLanguageParticipant of jsonLanguageParticipants) {
+					const languageId = jsonLanguageParticipant.languageId;
+					if (typeof languageId === 'string') {
+						languages.add(languageId);
+						if (jsonLanguageParticipant.comments === true) {
+							comments.add(languageId);
+						}
+					}
+				}
+			}
+		}
+		return !isEqualSet(languages, oldLanguages) || !isEqualSet(oldLanguages, oldComments);
```

Comment:
Should be `comments` instead of `oldLanguages`

Reply:
is also a bug in the html pr https://github.com/angelozerr/vscode/blob/290f9196d6a1aad30240b8b1ce32b2c407baa743/extensions/html-language-features/client/src/languageParticipants.ts#L58
https://github.com/microsoft/vscode/pull/202198